pandas_transform/pandas_transform.md

```diff
@@ -0,0 +1,995 @@
+
+If your original data is not already in a separate file you can use `.to_csv(data.csv)` to create a new file containing your DataFrame as a csv file. You can also create files in [many other formats](https://pandas.pydata.org/docs/reference/io.html).
+</div>
+
+### Create new columns
+
+@sage
+<div class="python_data_init">
+<lia-keep>
+<script type="text/x-sage">
+import pandas as pd
+covid_testing = pd.read_csv('https://raw.githubusercontent.com/arcus/education_modules/embedded_code/a_sample_module_template/covid_testing.csv')
+</script>
+</lia-keep>
+</div>
+<br>
+
+In `pandas` you can create a new column by calling your new column with the `.loc[:,"new_column"]` method and using a single equals sign `=` to define its contents.
+
+The next bit of code creates a new column named `new_column` in the DataFrame `covid_testing` in which every entry is the number 1:
+
+<div class="python_data">
+<lia-keep>
+<script type="text/x-sage">
+covid_testing.loc[:,"new_column"] = 1
+print(covid_testing.loc[:,"new_column"])
+</script>
+</lia-keep>
+</div>
+<br>
+
+Your new column can also depend on data in columns that already exist. Remember that each column, or series, needs to be called using the format `dataframe.loc[:,"column_name"]`.
+
+For example maybe you need to have patient ages recorded in months instead of years:
+
+<div class="python_data">
+<lia-keep>
+<script type="text/x-sage">
+covid_testing.loc[:,"age_months"] = covid_testing.loc[:,"age"]*12
+print(covid_testing.loc[:,["age","age_months"]])
+</script>
+</lia-keep>
+</div>
+<br>
+
+Or maybe you want a new column that displays the full name of each patient, rather than separate columns for first and last names.
+
+<div class="python_data">
+<lia-keep>
+<script type="text/x-sage">
+covid_testing.loc[:,"full_name"] = covid_testing.loc[:,"first_name"]+" " +covid_testing.loc[:,"last_name"]
+print(covid_testing.loc[:,["first_name", "last_name", "full_name"]])
+</script>
+</lia-keep>
+</div>
+
+
+
+### Edit existing columns
+@sage
+<div class="python_data_init">
+<lia-keep>
+<script type="text/x-sage">
+import pandas as pd
+covid_testing = pd.read_csv('https://raw.githubusercontent.com/arcus/education_modules/embedded_code/a_sample_module_template/covid_testing.csv')
+</script>
+</lia-keep>
+</div>
+
+You can make changes to existing entries using the same method and simply using the key that already exists for that column.
+
+What if you wanted to reformat the gender column to use `M` and `F` instead of spelling out male and female?
+
+**Using `=` to assign values**
+
+One way to approach this is to define a condition testing each row for whether the entry in the `gender` column is `male`. Then we can use that condition to change the entry to `M` if the condition is met.
+
+<div class="python_data_init">
```

Comment:
I think this should be a python_data block, right?

Reply:
yup!